src/core/wordle/api.js

```diff
@@ -0,0 +1,92 @@
+// Database access operations
+const path = require('path');
+const JSONdb = require('simple-json-db');
+
+/**
+ * @return {JSONdb}
+ */
+function initDB() {
+  const file = path.resolve(__dirname, 'db.json');
+  return new JSONdb(file);
+}
+
+/**
+ * @return {Array}
+ */
+function getGames() {
+  const db = initDB();
+  return db.get('games') ?? [];
+}
+
+/**
+ * @param {Object} game
+ */
+function addGame(game) {
+  const db = initDB();
+  const games = db.get('games');
+  games.push(game);
+  db.set('games', games);
+}
+
+/**
+ * @return {Object}
+ */
+function getSession() {
+  const db = initDB();
+  return db.get('session');
+}
+
+/**
+ * @return {Array}
+ */
+function getSessionScores() {
+  const session = getSession();
+  return session.scores ?? [];
+}
+
+/**
+ * @return {String}
+ */
+function getSessionChannelId() {
+  const session = getSession();
+  return session.channel_id;
+}

```

Comment:
Might be worth adding a few words about the uniqueness of the session, and/or maybe make sure that `channel_id`  can't be set unless reset to null first? To avoid accidental override

Reply:
In `startwordle` command, and befaore creating a new session (which means assigning a new value to `channel_id`), it's checked before whether an active session exists (`channel_id` is not null).
```js
    // Check if there is no active session, return if there is
    if (activeWordleSession()) {
      await interaction.reply(ephemeral(`A wordle session is already active, `+
       `make sure to end it first !`));
      return;
    }
    // Get the id of the channel where the command was sent
    const channelId = interaction.channel.id;
    createWordleSession(channelId);
```
You think the test should be added to the `api.js` as well? My approach was to keep `api.js` for simple data access operations, and to hadle the game logic in `core/wordle/index.js` and the commands.

Reply:
not necessarily what's under api.js as it's used by the rest of the code. Covering different use cases of index.js should cover the rest of the api